kubelet: add setting for configuring cpuManagerPolicy (#1638)
bottlerocket-os/bottlerocket: README.md

* `settings.kubernetes.kube-api-burst`: The burst to allow while talking with kubernetes.
* `settings.kubernetes.container-log-max-size`: The maximum size of container log file before it is rotated.
* `settings.kubernetes.container-log-max-files`: The maximum number of container log files that can be present for a container.

> * `settings.kubernetes.cpu-manager-policy`: Specifies the CPU manager policy. Possible values are `static` and `none`. Defaults to `none`. If you want to allow

Reviewer: nit: We usually wrap at the sentence break. This seems like kind of a wide wrapping.

pods with certain resource characteristics to be granted increased CPU affinity and exclusivity on the node, you can set this setting to `static`. We recommend
a full reboot if you change this setting after launch.

> * `settings.kubernetes.cpu-manager-reconcile-period`: Specifies the CPU manager reconcile period, which can configure CPU Manager Reconcile Kubelet CPU memory

Reviewer 2: ```suggestion
* `settings.kubernetes.cpu-manager-reconcile-period`: Specifies the CPU manager reconcile period, which controls how often updated CPU assignments are written to cgroupfs. 
```

* `settings.kubernetes.kube-api-burst`: The burst to allow while talking with kubernetes.
* `settings.kubernetes.container-log-max-size`: The maximum size of container log file before it is rotated.
* `settings.kubernetes.container-log-max-files`: The maximum number of container log files that can be present for a container.

> * `settings.kubernetes.cpu-manager-policy`: Specifies the CPU manager policy. Possible values are `static` and `none`. Defaults to `static`.

Reviewer 2: I'd like to default to "none" here given known incompatibilities with fluent-bit as it's deployed today. We can always change the default for future Kubernetes variants.

* `settings.kubernetes.kube-api-burst`: The burst to allow while talking with kubernetes.
* `settings.kubernetes.container-log-max-size`: The maximum size of container log file before it is rotated.
* `settings.kubernetes.container-log-max-files`: The maximum number of container log files that can be present for a container.

> * `settings.kubernetes.cpu-manager-policy`: Specifies the CPU manager policy. Possible values are `static` and `none`. Defaults to `none`.

Reviewer 3: We should probably document that Kubernetes strongly recommends a full reboot if you change this setting after launch.  Our kubernetes service and kubelet-config configuration-file will notice the update and be rewritten, but kubelet isn't designed to handle that change while the system is running, and things could start to go wrong when it notices the state file's policy doesn't match the config file's policy.

> * `settings.kubernetes.cpu-manager-reconcile-period`: Specifies the CPU manager reconcile period.

Reviewer 3: I think it'd help to give a brief explanation of why the user would care about changing these settings; they're pretty opaque, even in the k8s documentation.

Author: That sounds good. Thanks

> * `settings.kubernetes.cpu-manager-policy`: Specifies the CPU manager policy. Possible values are `static` and `none`. Defaults to `none`. If you want to allow pods with certain resource characteristics to be granted increased CPU affinity and exclusivity on the node, you can set this setting to `static`. We recommend a full reboot if you change this setting after launch.

Reviewer 3: ```suggestion
* `settings.kubernetes.cpu-manager-policy`: Specifies the CPU manager policy. Possible values are `static` and `none`. Defaults to `none`. If you want to allow pods with certain resource characteristics to be granted increased CPU affinity and exclusivity on the node, you can set this setting to `static`. You should reboot if you change this setting after startup - try `apiclient reboot`.
```

> * `settings.kubernetes.cpu-manager-reconcile-period`: Specifies the CPU manager reconcile period, which controls how often updated CPU assignments are written to cgroupfs.

Reviewer 3: ```suggestion
* `settings.kubernetes.cpu-manager-reconcile-period`: Specifies the CPU manager reconcile period, which controls how often updated CPU assignments are written to cgroupfs.  The value is a duration like `30s` for 30 seconds or `1h5m` for 1 hour and 5 minutes.
```
